docs/how-to/rocks/chisel-existing-rock.rst

```diff
@@ -0,0 +1,132 @@
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:run-rock]
+    :end-before: [docs:run-rock-end]
+    :dedent: 2
+
+This rock will also have other utilities like ``bash`` and ``apt`` which come
+from the rock's underlying Ubuntu 22.04 base.
+
+Rebuild the rock with a ``bare`` base
+-------------------------------------
+
+When starting to prepare the rock for production, the main goal is to get rid
+of all the software that is not necessary at runtime, and the first step
+towards achieving that goal is to use a ``bare``.
+
+In a separate directory, copy the *rockcraft.yaml* from above and replace
+``base: ubuntu@22.04`` with ``base: bare``. With this change, you must also
+remember to tell Rockcraft which Ubuntu release to use for its build
+environment. Do this by adding ``build-base: ubuntu@22.04`` to the
+*rockcraft.yaml* file, which should now look like this:
+
+.. literalinclude:: ../code/chisel-existing-rock/bare-rock/rockcraft.yaml
+    :language: yaml
+
+Pack this rock with the same ``rockcraft`` command as above:
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:pack-bare-rock]
+    :end-before: [docs:pack-bare-rock-end]
+    :dedent: 2
+
+This new rock (``bare-python_3.11_amd64.rock``) will now have about **28MB** -
+a ~33% size reduction - and also have a functional Python3.11 interpreter.
+Run the same "Hello, world" Python script as before to confirm:
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:run-bare-rock]
+    :end-before: [docs:run-bare-rock-end]
+    :dedent: 2
+
+
+The question then is: *how is rockcraft able to produce an equally functional
+Python rock with such a drastic reduction in size?*
+
+And the answer is: *the rock no longer has the Ubuntu base as its first layer,
+and thus no longer has utilities like ``bash`` and ``apt`` (which aren't needed
+for this use case anyway)!*
+
+Chisel the Python rock
+----------------------
+
+The last step in your way to production is to chisel the rock in order to
+further reduce its size and attack surface, by stripping down the internal
+Python components that are not needed for the final application.
+
+In this example, the Python application is a very simple "Hello, world", which
+means we need nothing else but the core Python modules. For that, copy the
+previous *rockcraft.yaml* file to a new directory and simply append the desired
+package slice names to the list of ``stage-packages``, like this:
+
+.. literalinclude:: ../code/chisel-existing-rock/chiselled-rock/rockcraft.yaml
+    :language: yaml
+
+Pack it with
+
+.. literalinclude:: ../code/chisel-existing-rock/task.yaml
+    :language: bash
+    :start-after: [docs:pack-chiselled-rock]
+    :end-before: [docs:pack-chiselled-rock-end]
+    :dedent: 2
+
+and the end result will be an astoundingly small Python rock with **13MB**!
```

Comment:
```suggestion
And the end result will be an astoundingly small Python rock with **13MB**!
```
(same reason)

Reply:
this is on purpose. I have a love & hate relationship with code blocks, cause in theory, there's no period  and thus no end of sentence. The scheme I was using was: _if the code block is preceded by `:`, then the next sentence is capitalized. otherwise it shall just continue in lower case_

For the sake of moving this PR, I've just accepted all changes